python/llm/example/GPU/HuggingFace/LLM/chatglm3/generate.py

```diff
@@ -20,24 +20,34 @@
 import numpy as np
 
 from ipex_llm.transformers import AutoModel
-from transformers import AutoTokenizer
 
 # you could tune the prompt based on your own model,
 # here the prompt tuning refers to https://github.com/THUDM/ChatGLM3/blob/main/PROMPT.md
 CHATGLM_V3_PROMPT_FORMAT = "<|user|>\n{prompt}\n<|assistant|>"
 
 if __name__ == '__main__':
     parser = argparse.ArgumentParser(description='Predict Tokens using `generate()` API for ChatGLM3 model')
-    parser.add_argument('--repo-id-or-model-path', type=str, default="THUDM/chatglm3-6b",
+    parser.add_argument('--repo-id-or-model-path', type=str, 
                         help='The huggingface repo id for the ChatGLM3 model to be downloaded'
                              ', or the path to the huggingface checkpoint folder')
     parser.add_argument('--prompt', type=str, default="AI是什么？",
                         help='Prompt to infer')
     parser.add_argument('--n-predict', type=int, default=32,
                         help='Max tokens to predict')
+    parser.add_argument('--modelscope', action="store_true", default=False, 
+                        help="Use models from modelscope")
 
     args = parser.parse_args()
-    model_path = args.repo_id_or_model_path
+
+    if args.modelscope:
+        from modelscope import AutoTokenizer
+        model_hub = 'modelscope'
+    else:
+        from transformers import AutoTokenizer
+        model_hub = 'huggingface'
+    
+    model_path = args.repo_id_or_model_path if args.repo_id_or_model_path else \
+        ("ZhipuAI/chatglm3-6b" if args.modelscope else "THUDM/chatglm3-6b")
```

Comment:
What if the user input other chatglm3 model.

Please do this only if args.repo_id_or_model_path is None.